### ISSUE TITLE
[FEATURE REQUEST] allow for multiple charts to be opened side-by-side

### ISSUE THREAD

**Issue by rehtlaw** [PerryAsleep/GrooveAuthor#11]
This came up in a discussion very recently, but the ability to put up two different charts side by side (maybe even from different game types) from the same simfile that automatically scroll together would be great. In theory you can already just open two instances of the editor side-by-side with different charts open, but it would be a bit QoL gain if both were synced up and used the same playhead/scroll position.

**Comment by PerryAsleep** [comment]
Thanks for writing this up. I agree this would be nice. Though to be honest I am not sure when I will have time for it. It would be a fairly large task. At a high level some of the problems are:
- Two charts can have different music. Almost no songs do this, but it is supported. Would probably just need to disable multi-chart viewing in this case.
- Two charts can have different timing events. Probably not an issue.
- Would need handling for which chart has focus. It probably doesn't make sense for many actions to work on more than one chart at a time. For example if you drag to select notes with an area that encompasses notes from two charts it probably doesn't make to select notes from both. Or if you press a key to add a new note it probably makes sense for that to apply to only one chart.
- Lots of things currently operate on the one active chart.
- Pretty minor but I'd want to consider the perf ramifications.

My first thought to handle all this would probably be to have one chart be the main chart and the other open one(s) be auxiliary charts that are just rendered but don't handle input. I think that would avoid most of the complications. But I'd need to give more thought to what the UX would be for handling all that.

Actually having written all that out it may really not be as bad as I initially thought. I'll keep thinking about this. Thanks for the suggestion.

**Comment by rehtlaw** [comment]
Yeah you could probably get away implementing the 2nd chart as a view-only window, but implementing proper window focus and letting both sides be edited would certainly be more intuitive for users.

**Comment by PerryAsleep** [comment]
This is now implemented and available in in [v0.3.0](https://github.com/PerryAsleep/GrooveAuthor/releases/tag/v0.3.0)
